examples/gradient.jl

```diff
@@ -56,3 +56,27 @@ gradient(f, inputs, cfg)
 gradient!(results, f, inputs)
 
 gradient(f, inputs)
+
+"""
+    ReverseDiff.makeGradients(f, input :: Vector{<:Real})
+
+Returns `(∇f!, f∇f!, g, yg)`.
+
+`∇f!` takes a value similar to `input`, and returns the gradient at that point.
+This gradient is also written to `g` to allow more efficient use that avoids memory allocation.
+
+`f∇f!` takes a value similar to `input`, and returns both the function value and the gradient at that point.
+This pair (function value, gradient) is also written to `yg` to allow more efficient use that avoids memory allocation.
+
+Note: THIS USE CASE IS NOT THREAD SAFE.
+"""
+makeGradients(f, x0) = begin
+  const f_tape = GradientTape(f, x0)
+  const compiled_f_tape = compile(f_tape)
+  g = similar(x0)
+  yg = DiffBase.GradientResult(g)
+  cfg = GradientConfig(x0)
+  ∇f!(x)  = gradient!(g, compiled_f_tape, x)
+  f∇f!(x) = gradient!(yg, compiled_f_tape, x)
+  return(∇f!, f∇f!, g, yg)
+end
```

Comment:
Apologies for the late response! Could you change the above to:

```julia
function makeGradients(f, x0)
    f_tape = GradientTape(f, x0)
    compiled_f_tape = compile(f_tape)
    g = similar(x0)
    yg = DiffBase.GradientResult(g)
    cfg = GradientConfig(x0)
    ∇f! = x -> gradient!(g, compiled_f_tape, x)
    f∇f! = x -> gradient!(yg, compiled_f_tape, x)
    return(∇f!, f∇f!, g, yg)
end
```

This form...
- ...removes the `const`, which doesn't mean anything for locally scoped variables
- ...puts a space between `return` and the output tuple, to ensure it doesn't get confused for a function call
- ...uses anonymous function syntax instead of method definition syntax (avoiding potential confusion w.r.t. locally-scoped method overloading)